hasher-matcher-actioner/hmalib/lambdas/api/datasets.py

```diff
@@ -294,4 +339,66 @@ def delete_dataset(key=None) -> DeleteDatasetResponse:
         hmaconfig.delete_config(config)
         return DeleteDatasetResponse(response="The privacy group is deleted")
 
+    @datasets_api.get("/match-settings", apply=[jsoninator])
+    def get_all_match_settings() -> MatchSettingsResponse:
+        """
+        Return all match settings configs
+        """
+        return MatchSettingsResponse(
+            match_settings=[
+                MatchSettingsResponseBody(c)
+                for c in AdditionalMatchSettingsConfig.get_all()
+            ]
+        )
+
+    @datasets_api.get("/match-settings/<key>", apply=[jsoninator])
+    def get_match_settings(
+        key=None,
+    ) -> MatchSettingsResponseBody:
+        """
+        Return a match settings config for a given privacy_group_id
+        """
+        if config := AdditionalMatchSettingsConfig.get(str(key)):
+            return MatchSettingsResponseBody(config)
+        return bottle.abort(400, f"No match_settings for pg_id {key} found")
+
+    @datasets_api.post(
+        "/match-settings", apply=[jsoninator(MatchSettingsUpdateRequest)]
+    )
+    def create_or_update_match_settings(
+        request: MatchSettingsUpdateRequest,
+    ) -> MatchSettingsUpdateResponse:
+        """
+        Create or update a match settings config for a given privacy_group_id
+        """
+        if config := AdditionalMatchSettingsConfig.get(request.privacy_group_id):
+            config.pdq_match_threshold = request.pdq_match_threshold
+            hmaconfig.update_config(config)
+
+            return MatchSettingsUpdateResponse(
+                f"match_settings updated for pg_id {request.privacy_group_id} with pdq_match_threshold={request.pdq_match_threshold}"
```

Comment:
nit: Personally, don't see the value of logging style messages in responses. A 200 is enough if you don't have anything to return.